corehq/apps/domain/views.py

```diff
@@ -1255,12 +1255,10 @@ def create_form(form_class):
     @property
     @memoized
     def select_subscription_type_form(self):
-        if self.request.method == 'POST':
-            for form_slug in self.slug_to_form:
```

Comment:
Looked like the previous code looped through all possible forms and returned when it found a matching key in the request, but the request didn't contain any of those keys, and instead uses a 'slug' key. Tested out saving the form successfully and attempting to save it with errors.